examples/server/server.c

```diff
@@ -683,6 +683,58 @@ static void SetKeyShare(WOLFSSL* ssl, int onlyKeyShare, int useX25519,
                                 XSTRLEN("KYBER90S1024")) == 0) {
                 groups[count] = WOLFSSL_KYBER90S1024;
             }
+            else if (XSTRNCMP(oqsAlg, "P256_NTRUHPS2048509",
```

Comment:
How does using strncmp, and strlen on a constant string help protect anything here?

             else if (XSTRNCMP(oqsAlg, “P521_NTRUHPS4096821”,
                                 XSTRLEN(“P521_NTRUHPS4096821")) == 0) {
                 groups[count] = WOLFSSL_P521_NTRU_HPS4096821;
             }

strncmp, I thought, used the length to be the length of the not-constant part, the buffer that you might read past the bounds on.

Reply:
Given it is following the existing pattern, this is being overly picky. Our client and server examples should probably do a better job checking that the command line arguments are within the ballpark of the correct length and then not have to worry about this.